target/linux/mediatek/dts/mt7622-ruijie-rg-ew3200gx-pro.dts

```diff
@@ -0,0 +1,323 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+/dts-v1/;
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/gpio/gpio.h>
+
+#include "mt7622.dtsi"
+#include "mt6380.dtsi"
+
+/ {
+	model = "Ruijie RG-EW3200GX PRO";
+	compatible = "ruijie,rg-ew3200gx-pro", "mediatek,mt7622";
+
+	aliases {
+		ethernet0 = &gmac0;
+		label-mac-device = &gmac0;
+		led-boot = &led_system;
+		led-failsafe = &led_system;
+		led-running = &led_system;
+		led-upgrade = &led_system;
+		serial0 = &uart0;
+	};
+
+	chosen {
+		stdout-path = "serial0:115200n1";
+		bootargs = "console=ttyS0,115200n1 swiotlb=512";
+	};
+
+	cpus {
+		cpu@0 {
+			proc-supply = <&mt6380_vcpu_reg>;
+			sram-supply = <&mt6380_vm_reg>;
+		};
+
+		cpu@1 {
+			proc-supply = <&mt6380_vcpu_reg>;
+			sram-supply = <&mt6380_vm_reg>;
+		};
+	};
+
+	gpio-keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&pio 0 GPIO_ACTIVE_LOW>;
+		};
+
+		wps {
+			label = "wps";
+			linux,code = <KEY_WPS_BUTTON>;
+			gpios = <&pio 102 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	gpio-leds {
+		compatible = "gpio-leds";
+
+		mesh_green {
+			label = "green:mesh";
+			gpios = <&pio 79 GPIO_ACTIVE_LOW>;
+		};
+
+		mesh_red {
+			label = "red:mesh";
+			gpios = <&pio 82 GPIO_ACTIVE_LOW>;
+		};
+
+		led_system: system_blue {
+			label = "blue:system";
+			gpios = <&pio 81 GPIO_ACTIVE_LOW>;
+			default-state = "on";
+		};
+	};
+
+	memory {
+		reg = <0 0x40000000 0 0xf000000>;
```

Comment:
Using `0xf000000` (== 240M) looks weird to me and memory autodetection is working fine for other devices with a total of 256M. Is that necessary due to use of the upper 16M by something else (ie. TF-A or OP-TEE)? If so, a `reserved-memory` node would be a more appropriate way to express that.

Reply:
I copied the `memory` nodes from the original DTS directly. Memory auto-recognition worked fine for this device. Maybe I can delete the `memory` node?

Reply:
I see that other boards have
```
reg = <0 0x40000000 0 0x40000000>;
```
which should enable memory size auto-detection with the full addressable range of the CPU. As this is the same for all boards, we could as well remove it from the board-specific DTS and move it to `mt7622.dtsi` instead.
Anyway, you will have to try and see if it works (ie. boots successfully with full amount of memory detected and usable, no overlap with ranges used by TF-A or OP-TEE)

Reply:
I changed the range in `memory` node. It was tested and works fine. The range of memory automatically detected has the same size as the original image. 
It looks like that the range of memory used by TF-A has been included in [mt7622.dtsi](https://github.com/torvalds/linux/blob/fb184c4af9b9f4563e7a126219389986a71d5b5b/arch/arm64/boot/dts/mediatek/mt7622.dtsi#L130)😀.